File: src/main/java/sk/hudak/prco/dto/error/ErrorCreateDto.java
package sk.hudak.prco.dto.error;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import sk.hudak.prco.api.ErrorType;
import sk.hudak.prco.api.EshopUuid;
import sk.hudak.prco.dto.DtoAble;

@Setter
@Getter
@Builder
public class ErrorCreateDto implements DtoAble {

    private EshopUuid eshopUuid;
    private ErrorType errorType;
    private String statusCode;
    //TODO impl validation cez validation spring framework a moje vynimky
//    @NotNull
    private String message;
    private String fullMsg;
    private String url;
    private String additionalInfo;
}

File: src/main/java/sk/hudak/prco/dto/error/ErrorCreateDto.kt
package sk.hudak.prco.dto.error

import sk.hudak.prco.api.ErrorType
import sk.hudak.prco.api.EshopUuid
import sk.hudak.prco.dto.DtoAble

data class ErrorCreateDto(val eshopUuid: EshopUuid? = null,
                          val errorType: ErrorType? = null,
                          val statusCode: String? = null,
                          var message: String? = null,
                          var fullMsg: String? = null,
                          val url: String? = null,
                          val additionalInfo: String? = null) : DtoAble
